### PR TITLE
Move login modal to component

### PR DESCRIPTION
**Pull request by danawoodman**
In an attempt to start using components, I converted the login modal to a component

### CODE REVIEW

File: frontend/simple/sass/_groupincome.scss [hunk 1]
```diff
@@ -74,3 +74,22 @@
...
+}
+.is-warning {
+  color: $orange;
+}
+.is-light {
+  color: $grey;
+}
+.is-small {
+  font-size: $size-7;
+}
```
Comment (taoeffect) [inline frontend/simple/sass/_groupincome.scss:95]: These look like they're just copied from Bulma? Why are they necessary here?

If they are necessary for some reason, can you add that explanation in a comment to the source as well?

File: frontend/simple/views/NavBar.vue [hunk 2]
```diff
@@ -10,27 +10,60 @@
...
           <span class="nav-item is-tab control">
             <router-link id="SignupBtn" v-show="!$store.state.loggedIn" class="button is-success" to="signup"><i18n>Sign Up</i18n></router-link>
-            <a href="#" id="LoginBtn" class="button"
-               v-bind:class="$store.state.loggedIn  ? 'is-danger' : 'is-primary'"
-              @click.prevent="toggleModal"
+            <a
+              class="button is-primary"
+              href="#"
+              id="LoginBtn"
+              v-if="!$store.state.loggedIn"
```
Comment (taoeffect) [inline frontend/simple/views/NavBar.vue:61]: This conditional is the exact same condition, just reversed, below. Use `v-else` then.

File: frontend/simple/views/NavBar.vue [hunk 3]
```diff
@@ -39,44 +72,7 @@
...
-                  <i18n>Login</i18n>
-                </button>
-              </p>
-            </div>
-          </div>
-        </div>
-      </div>
-      <div class="modal-close" @click="toggleModal"></div>
-    </div>
+    <login-modal :visible="loginModalVisible" />
```
Comment (taoeffect) [inline frontend/simple/views/NavBar.vue]: Neat-o!

File: frontend/simple/main.js [hunk 4]
```diff
@@ -14,14 +15,16 @@ Vue.use(VeeValidate)
...
-  }
+  const user = await db.loadCurrentUser()
+  if (user) await store.dispatch('login', user)
+
   /* eslint-disable no-new */
   new Vue({
     router: router,
-    components: {NavBar},
+    components: {
+      LoginModal,
```
Comment (taoeffect) [inline frontend/simple/main.js]: No need to add LoginModal here, it's added elsewhere (in NavBar).

**Comment by taoeffect** [comment]
Feedback done.

Overview:

- See comments
- Fix build failing
- Great job overall, and nice job making sure to i18n all of the text

File: frontend/simple/sass/_groupincome.scss (hunk 1 above)
Comment (danawoodman) [reply frontend/simple/sass/_groupincome.scss:95]: Sorta. Bulma does a poor job of documenting these features in the readme but as far as I can tell, they only apply to very specific components like buttons and not plain text. I can make a note about that.

File: frontend/simple/views/NavBar.vue (hunk 2 above)
Comment (danawoodman) [reply frontend/simple/views/NavBar.vue:61]: OK, good to know that exists 😄

File: frontend/simple/main.js (hunk 4 above)
Comment (danawoodman) [reply frontend/simple/main.js]: Make sense, will change

**Comment by danawoodman** [comment]
@taoeffect thanks! This is a WIP so I still have more work to be done but this is the first "component" in the app which is neat 😊

File: frontend/simple/views/NavBar.vue [hunk 5]
```diff
@@ -39,45 +79,11 @@
...
-          </div>
-        </div>
-      </div>
-      <div class="modal-close" @click="toggleModal"></div>
-    </div>
-    <time-travel v-show="showTimeTravel" :toggleVisibility="toggleTimeTravel"></time-travel>
+    <login-modal
+      v-if="loginModalVisible"
+      @close="closeLoginModal"
+    />
```
Comment (taoeffect) [inline frontend/simple/views/NavBar.vue:85]:
```js
if (line.width <= 100) {
    console.log('do not wrap line')
} else {
   console.log('wrap line')
}
```

:P

Can you run through this whole file and make sure everything follows that rule? We want to keep the size of the files small if possible.

Comment (danawoodman) [reply frontend/simple/views/NavBar.vue:85]: Personally, I prefer legibility over line length as we spend much more time reading code then writing. This format makes it easier to add/remove properties in bulk and makes it faster to parse what properties a component contains. It's somewhat a standard format in the React world. 

That said, if this isn't something you aren't open to, I can revert.

**Pull request merged**